blogposts/announcing-sourcegraph-3.1.md

```diff
@@ -0,0 +1,100 @@
+
+## Site admin improvements
+
+Administering Sourcegraph in an enterprise SSO environment is now easier thanks to a couple of small, but important features:
+
+- [Querying users by email address](https://sourcegraph.com/api/console#%7B%22query%22%3A%22%7B%5Cn%20%20user(email%3A%20%5C%22ryan%40sourcegraph.com%5C%22)%20%7B%5Cn%20%20%20%20id%5Cn%20%20%7D%5Cn%7D%5Cn%22%7D).
+- [Disabling the changing of usernames](https://docs.sourcegraph.com/admin/config/critical_config#reference).
+
+Making configuration changes through the JSON editor is now easier too with easy to access documentation in the editor itself.
+
+<!-- TODO (ryan): GIF showing editor using the help tool tip with an item that requires a choice of values -->
+
+## Improved configuration management and documentation
+
+Sourcegraph 3.0 introduced the [management console](https://about.sourcegraph.com/blog/sourcegraph-3.0#management-console-and-site-config-improvements) for editing critical configuration to allow recovery from misconfiguration in the site admin.
+
+We've [updated our documentation](https://docs.sourcegraph.com/admin/config) to reflect this change, separating configuration into two sections:
+
+- [Critical configuration](https://docs.sourcegraph.com/admin/config/critical_config).
+- [Site configuration](https://docs.sourcegraph.com/admin/config/site_config).
+
+Reference documentation for [site configuration](https://docs.sourcegraph.com/admin/config/site_config#reference) and [critical configuration](https://docs.sourcegraph.com/admin/config/critical_config#reference) values are now rendered using the schema files themselves to ensure they'll never get out of date.
+
+## New Sourcegraph extension APIs
+
+The [Sourcegraph extension API](https://docs.sourcegraph.com/extensions/authoring) continues to evolve with two new APIs:
+
+- [Exposing text selection to extensions](https://github.com/sourcegraph/sourcegraph/issues/702).
+- [APIs for listening to active text editor changes and selections](https://github.com/sourcegraph/sourcegraph/issues/2026).
+
+This means the `[Git extras Sourcegraph extension](https://sourcegraph.com/extensions?query=git-extras) works better than ever. Want this functionality on GitHub too? Just install the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension).
+
+<!-- TODO (ryan): Gif of git-extras extension -->
+
+## Switching to scheduled releases on the 20th of each month
+
+We received feedback from many of you asking for a more regular and predictable release schedule and as a result, Sourcegraph will now be released on the 20th of each month 🎉.
+
+## Changelog
+
+See the [Sourcegraph changelog](https://github.com/sourcegraph/sourcegraph/blob/master/CHANGELOG.md#31) for a list of all changes in this release.
```

Comment:
Why not inline that changelog here? Then ctrl+F/cmd+F on this blog post for a specific topic will work better.

Reply:
I disagree with inlining the changelog.

Reply:
Arguments:
- GitLab doesn't do it.
- It feels lazy and duplicative.
- The blog post should be a high quality summary and answer "why" we did things

Reply:
VS Code does it. Does that change your mind? 

Let’s do it anyway, I know times when people have become aware of things because they were inclined in the changelog here.

Reply:
Your call. If we leave it in, then I see that as an even stronger reason for the non-changelog section to be more of a "why" narrative instead of "what" (which I think is lacking in the current draft regardless).

Reply:
Let's leave it in for now and discuss for 3.2.